tests/budget_accounting_test.py

```diff
@@ -48,5 +53,147 @@ def test_compute_budgets(self):
         self.assertEqual(budget2.delta, 1e-6)
 
 
+class PLDBudgetAccountantTest(unittest.TestCase):
+
+    def test_noise_not_calculated(self):
+        with self.assertRaises(AssertionError):
+            mechanism = MechanismSpec(NoiseKind.LAPLACE)
+            print(mechanism.noise_standard_deviation())
+
+    def test_invalid_epsilon(self):
+        with self.assertRaises(ValueError):
+            PLDBudgetAccountant(total_epsilon=0, total_delta=1e-5)
+
+    def test_invalid_delta(self):
+        with self.assertRaises(ValueError):
+            PLDBudgetAccountant(total_epsilon=1, total_delta=-1e-5)
+
+    def test_compute_budgets_none_noise(self):
+        accountant = PLDBudgetAccountant(total_epsilon=3, total_delta=1e-5)
+        accountant.compute_budgets()
+        self.assertEqual(None, accountant.minimum_noise_std)
+
+    def test_compute_budgets(self):
+
+        @dataclass
+        class ComputeBudgetMechanisms:
+            count: int
+            expected_noise_std: float
+            noise_kind: NoiseKind
+            weight: float
+            sensitivity: float
+
+        @dataclass
+        class ComputeBudgetTestCase:
+            name: str
+            expected_epsilon: float
+            delta: float
+            expected_pipeline_noise_std: float
+            mechanisms: []
+
+        testcases = [
+            ComputeBudgetTestCase(name="laplace_mechanisms",
```

Comment:
Thanks for having tests in table, I realized that we need to cover additional cases.

Please add 2 additional tests, with `delta = 0`
1. 2 Laplace mechanisms, sensitivity=1, weight = 1 (with delta = 0, PLD should coincide with a standard eps-dp arithmetic)
2. 1 Gaussian mechanism (delta = 0, can't be Gaussian, so compute_budgets should raise Exception

Reply:
@dvadym I am not sure I handled correctly this calculation to take into account the weight and sensitivities per mechanism. I added a mention for the calculation up above.

Thanks for catching these cases!